### DISCUSSION TITLE
How to obtain the value in the ISymbol

### DISCUSSION THREAD

**Discussion by yz-qiang** [dotnet/roslyn#65336] [Q&A]
Hi,
To obtain the fully-qualified name of the method, I override the `visit()`, the written code as follow:
```
public override SyntaxNode VisitInvocationExpression(InvocationExpressionSyntax node)
        {
            Console.WriteLine("Node: " + node.ToString()); // output is: sourceTree.GetRoot()
            IEnumerable<SyntaxNode> Pre = node.DescendantNodesAndSelf();  
            string old_innode = node.Expression.GetFirstToken().ToString(); 
            Console.WriteLine(old_innode.ToString());

            foreach (SyntaxNode INnode_id in Pre)
            {
                if (INnode_id.Kind().ToString() == "IdentifierName")
                {
                    ISymbol INnode_Type = SemanticModel.GetSymbolInfo(INnode_id).Symbol;
                    var new_INnode = INnode_Type.ToString();  //  
                    WriteLine(new_INnode); // output is: sourceTree.  But, the expected output is Microsoft.CodeAnalysis.SyntaxTree
                } 
                WriteLine(INnode_id.Kind().ToString());
            }
            return node;
        }
```
Actually, when I debug this program, I can see `Microsoft.CodeAnalysis.SyntaxTree` value is stored in the `INnode_Type ` (as shown in the following figure).  But so far I haven't found a way to print this value. Excuse me, is there any way to output this value?  Thank you so much!
![image](https://user-images.githubusercontent.com/72866861/201121722-aba65338-7aa5-4ec0-a5ff-f75033f885ad.png)

**Comment by CyrusNajmabadi** [comment]
@yz-qiang use .Locations or .DeclaringSyntaxReferences.  also, def recommend discord for this.

**Reply by yz-qiang** [reply]
Use  `.Locations` or `.DeclaringSyntaxReferences` ? Can you share the specific implementation code with me? Please. Thank you so much.

**Reply by CyrusNajmabadi** [reply]
INode_Type.DeclaringSyntaxReferences. :-)

**Reply by jasonmalinowski** [reply]
@CyrusNajmabadi In this case I think @yz-qiang wants the type, so he'd have to grab ILocalSymbol.Type?

**Reply by CyrusNajmabadi** [reply]
Not sure. I'd recommend discord as it will make it easier to understand what is needed and how best to do it.

**Reply by yz-qiang** [reply]
> @CyrusNajmabadi In this case I think @yz-qiang wants the type, so he'd have to grab ILocalSymbol.Type?

Yes, I need  `sourceTree`  of the type, that is, `Microsoft.CodeAnalysis.SyntaxTree`.

Thanks for your suggestion. I replaced the code `ISymbol INnode_Type = SemanticModel.GetSymbolInfo(INnode_id).Symbol;` into the `ILocalSymbol INnode_Type = (ILocalSymbol)SemanticModel.GetSymbolInfo(INnode_id).Symbol;`. And then through the  `var new_INnode = INnode_Type.Type.ToString();`, I obtain the type. Thank you very much.
